src/Text/Parsing/StringParser/Combinators.purs

```diff
@@ -149,3 +150,10 @@ manyTill p end = scan
   scan = (end *> pure Nil) <|> do x <- p
                                   xs <- scan
                                   pure (Cons x xs)
+
+-- | Parse several phrases until the specified terminator matches, requiring at least one match.
+many1Till :: forall a end. Parser a -> Parser end -> Parser (List a)
+many1Till p end = do
+  x <- p
+  xs <- manyTill p end
```

Comment:
OK - I think I see what you mean - I'd never have spotted it.  I'll try it out - it certainly simplifies manyTill.